libleptongui/src/x_dialog.c

```diff
@@ -86,6 +86,33 @@ int generic_confirm_dialog (const char *msg)
 
 /***************** End of generic confirm dialog box *********************/
 
+/*! \brief Launch generic error dialog.
+ *  \par Function Description
+ *
+ *  Launches a generic error message dialog displaying one or two
+ *  messages.  The second message is optional and may be set to
+ *  NULL.
+ *
+ * \param [in] primary_message The main error message.
+ * \param [in] secondary_message The additional or detail message.
+ */
+void
+generic_error_dialog (const char *primary_message,
+                      const char *secondary_message)
+{
+  GtkWidget *dialog = gtk_message_dialog_new (NULL,
+                                              GTK_DIALOG_MODAL,
+                                              GTK_MESSAGE_ERROR,
+                                              GTK_BUTTONS_OK,
+                                              primary_message);
```

Comment:
Here I get a compiler warning:
```
x_dialog.c:107: warning:
format string is not a string literal
(potentially insecure) [-Wformat-security]
```
I managed to fix it this way:
```diff
diff --git a/libleptongui/src/x_dialog.c b/libleptongui/src/x_dialog.c
index 8d3106dd8..672e102e3 100644
--- a/libleptongui/src/x_dialog.c
+++ b/libleptongui/src/x_dialog.c
@@ -104,7 +104,8 @@ generic_error_dialog (const char *primary_message,
                                               GTK_DIALOG_MODAL,
                                               GTK_MESSAGE_ERROR,
                                               GTK_BUTTONS_OK,
-                                              primary_message);
+                                              "%s",
+                                              primary_message ? primary_message : "");
   g_object_set (G_OBJECT (dialog),
                 "secondary-text", secondary_message,
                 NULL);
```

Reply:
While I don't see such a warning when compiled with *gcc* or *g++*, here is my fix which seems to be more straightforward:
[0001-gui-Fix-the-warning-format-string-is-not-a-string-li.patch.txt](https://github.com/lepton-eda/lepton-eda/files/9690542/0001-gui-Fix-the-warning-format-string-is-not-a-string-li.patch.txt)

If you like it, I'll push it, otherwise, please commit your solution yourself, it also works OK for me :-)